ambry-cloud/src/main/java/com.github.ambry.cloud/VcrServer.java

```diff
@@ -0,0 +1,246 @@
+      this.metrics = new VcrMetrics(registry);
+      reporter = JmxReporter.forRegistry(registry).build();
+      reporter.start();
+
+      logger.info("creating configs");
+      NetworkConfig networkConfig = new NetworkConfig(properties);
+      StoreConfig storeConfig = new StoreConfig(properties);
+      ServerConfig serverConfig = new ServerConfig(properties);
+      ReplicationConfig replicationConfig = new ReplicationConfig(properties);
+      CloudConfig cloudConfig = new CloudConfig(properties);
+      ConnectionPoolConfig connectionPoolConfig = new ConnectionPoolConfig(properties);
+      ClusterMapConfig clusterMapConfig = new ClusterMapConfig(properties);
+      if (sslConfig == null) {
+        sslConfig = new SSLConfig(properties);
+      }
+      // verify the configs
+      properties.verify();
+
+      // Note: using static cluster until Helix version implemented
+      if (virtualReplicatorCluster == null) {
+        // TODO: obtain factory from config property
+        virtualReplicatorCluster = new StaticVcrCluster(cloudConfig, clusterMapConfig, clusterMap);
+      }
+
+      // initialize cloud destination
+      if (cloudDestinationFactory == null) {
+        cloudDestinationFactory = Utils.getObj(cloudConfig.cloudDestinationFactoryClass, properties, registry);
+      }
+
+      scheduler = Utils.newScheduler(serverConfig.serverSchedulerNumOfthreads, false);
+      StoreKeyFactory storeKeyFactory = Utils.getObj(storeConfig.storeKeyFactory, clusterMap);
+      FindTokenFactory findTokenFactory = Utils.getObj(replicationConfig.replicationTokenFactory, storeKeyFactory);
+
+      connectionPool = new BlockingChannelConnectionPool(connectionPoolConfig, sslConfig, clusterMapConfig, registry);
+      connectionPool.start();
+
+      StoreKeyConverterFactory storeKeyConverterFactory =
+          Utils.getObj(serverConfig.serverStoreKeyConverterFactory, properties, registry);
+
+      cloudBackupManager =
+          new CloudBackupManager(cloudConfig, replicationConfig, clusterMapConfig, storeConfig, storeKeyFactory,
+              clusterMap, virtualReplicatorCluster, cloudDestinationFactory, scheduler, connectionPool, registry,
+              notificationSystem, storeKeyConverterFactory, serverConfig.serverMessageTransformer);
+      cloudBackupManager.start();
+
+      DataNodeId currentNode = virtualReplicatorCluster.getCurrentDataNodeId();
+      ArrayList<Port> ports = new ArrayList<Port>();
+      ports.add(new Port(networkConfig.port, PortType.PLAINTEXT));
+      if (currentNode.hasSSLPort()) {
+        ports.add(new Port(cloudConfig.vcrSslPort, PortType.SSL));
+      }
+      networkServer = new SocketServer(networkConfig, sslConfig, registry, ports);
+      networkServer.start();
+
+      // TODO: for recovery, need AmbryRequests and RequestHandlerPool
+
+      List<AmbryHealthReport> ambryHealthReports = new ArrayList<>();
+      clusterParticipant.participate(ambryHealthReports);
```

Comment:
why are we trying to participate in the main cluster?

Reply:
Good catch.  This was copied from AmbryServer and I neglected to remove it.